LeaderboardBackend/Services/Impl/AccountConfirmationService.cs

```diff
@@ -0,0 +1,65 @@
+using LeaderboardBackend.Models.Entities;
+using NodaTime;
+
+namespace LeaderboardBackend.Services;
+
+public class AccountConfirmationService : IAccountConfirmationService
+{
+    private readonly ApplicationContext _applicationContext;
+    private readonly IEmailSender _emailSender;
+    private readonly IClock _clock;
+
+    public AccountConfirmationService(ApplicationContext applicationContext, IEmailSender emailSender, IClock clock)
+    {
+        _applicationContext = applicationContext;
+        _emailSender = emailSender;
+        _clock = clock;
+    }
+
+    public async Task<AccountConfirmation?> GetConfirmationById(Guid id)
+    {
+        return await _applicationContext.AccountConfirmations.FindAsync(id);
+    }
+
+    public async Task<CreateConfirmationResult> CreateConfirmationAndSendEmail(User user, CancellationToken token = default)
+    {
+        if (user.Role is not UserRole.Registered)
+        {
+            return new BadRole();
+        }
+
+        Instant now = _clock.GetCurrentInstant();
+
+        AccountConfirmation newConfirmation =
+            new()
+            {
+                CreatedAt = now,
+                ExpiresAt = now + Duration.FromHours(1),
+                UserId = user.Id,
+            };
+
+        _applicationContext.AccountConfirmations.Add(newConfirmation);
+
+        await _applicationContext.SaveChangesAsync(token);
+
+        try
+        {
+            await _emailSender.EnqueueEmailAsync(
+                user.Email,
+                "Confirm Your Account",
+                GenerateAccountConfirmationEmailBody(user, newConfirmation)
+            );
+        }
+        catch
+        {
+            // TODO: Log/otherwise handle the fact that the email failed to be queued - zysim
+            return new EmailFailed();
+        }
+
+        return newConfirmation;
+    }
+
+    // TODO: Finalise message contents
+    private string GenerateAccountConfirmationEmailBody(User user, AccountConfirmation confirmation) =>
+        $@"Hi {user.Username},<br/><br/>Click <a href=""/confirm-account?code={Convert.ToBase64String(confirmation.Id.ToByteArray())}"">here</a> to confirm your account.";
```

Comment:
You need to put the full website URL, not just the path; the user will be opening this link from their mail client. Also, use this encoding for GUIDs to ensure url safety and to remove padding: https://datatracker.ietf.org/doc/html/rfc7515#page-55. I suggest adding an extension method for it.

Reply:
I remember reading somewhere in the .NET docs that paths will be relative to the base URL. Of course there's no way to really prove it until we do start sending emails.

I'll switch to the full path, and I'll implement the extension method.

Reply:
> I remember reading somewhere in the .NET docs that paths will be relative to the base URL

Whatever that was, there is no way it could apply to emails.

Gonna need to add required configuration for the frontend base URL. In [AppConfig](https://github.com/leaderboardsgg/leaderboard-backend/blob/main/LeaderboardBackend/AppConfig.cs#L6) seems good.